script.js

```diff
@@ -0,0 +1,62 @@
+function capitalize(string) {
+  return string.charAt(0).toUpperCase() + string.slice(1);
+}
+
+function reverseString(string) {
+  return string.split("").reverse().join("");
+}
+
+const calculator = {
+  add: function (a, b) {
+    return a + b;
+  },
+  subtract: function (a, b) {
+    return a - b;
+  },
+  divide: function (a, b) {
+    return a / b;
+  },
+  multiply: function (a, b) {
+    return a * b;
+  },
+};
+
+function caesarCipher(string, shift) {
+  const inputInArray = string.split("");
+
+  inputInArray.forEach(function (value, i) {
+    if (value.match(/^[a-zA-Z]*$/))
+      inputInArray[i] = nextCharacter(value, shift);
+  });
+
+  const shifted = inputInArray.join("");
+
+  return shifted;
+}
+
+function nextCharacter(character, shift) {
+  let nextUnicode = character.charCodeAt(0) + shift;
+  if (
+    (character.match(/^[A-Z]*$/) && nextUnicode > 90) ||
+    (character.match(/^[a-z]*$/) && nextUnicode > 122)
+  )
+    nextUnicode -= 26;
+  if (
+    (character.match(/^[A-Z]*$/) && nextUnicode < 65) ||
+    (character.match(/^[a-z]*$/) && nextUnicode < 97)
```

Comment:
You are doing the same match check multiple times. You can do the match check once at the beginning of the function, save the result in a variable and then reuse it.

Reply:
Ok, I'll look into it